tests/compile-fail/range.rs

```diff
@@ -0,0 +1,15 @@
+#![feature(step_by)]
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#[deny(range_step_by_zero)]
+
+fn main() {
+    (0..1).step_by(0); //~ERROR Range::step_by(0) may loop forever.
+    // No warning for non-zero step
+    (0..1).step_by(1);
+
+    // No warning: we only check for range literals
+    let x = 0..1;
+    x.step_by(0);
```

Comment:
Lints run after the typechecker, so we have type info available. If you check for method call of "step_by" and then match the type of "x" to be a Range, you can also catch this one.

Search for `expr_ty` in the codebase to find hints about this.